src/core/server/http/base_path_proxy_server.ts

```diff
@@ -143,12 +143,25 @@ export class BasePathProxyServer {
       handler: {
         proxy: {
           agent: this.httpsAgent,
-          host: this.server.info.host,
           passThrough: true,
-          port: this.devConfig.basePathProxyTargetPort,
-          // typings mismatch. h2o2 doesn't support "socket"
-          protocol: this.server.info.protocol as HapiProxy.ProxyHandlerOptions['protocol'],
           xforward: true,
+          mapUri: async (request) => {
+            return {
+              // Passing in the headers is a workaround until this is fixed:
+              // https://github.com/hapijs/h2o2/issues/124
+              headers:
+                request.headers['content-length'] != null
+                  ? { 'content-length': request.headers['content-length'] }
+                  : undefined,
+              uri: Url.format({
```

Comment:
Would you mind adding an integration test?

Reply:
I will see if I can get one added, appreciate the code review for this.

Reply:
Ok, we're in luck here. I figured out how to port all the tests from:
* src/core/server/http/http_server.test.ts

To `base_path_proxy_server.test.ts` as well as add additional custom tests for it. I also took my original code out and was able to watch it fail to double check that the tests would fail in a normal TDD fashion. I then put my code back and everything passes.